### PR TITLE
feat(Task-7): As a User, I want a minimalistic StartPage.

### PR DESCRIPTION
**Pull request by jLagosCarrera**

### CODE REVIEW

File: src/app/app.module.js [hunk 1]
```diff
@@ -4,14 +4,24 @@ import angularmd from 'angular-material';
...
+//Values
+import latestSearches from './shared/data/latestSearches';
+import fiveDayForecast from './shared/data/fiveDayForecast';
+//Components
+import homePage from './home-page/home-page';
+//Services
+import openWeatherMaps from './shared/services/openWeatherMaps';
+import latestSearchesUtil from './shared/services/latestSearchesUtil';
 
 export default angular.module('app', [uiRouter, angularanimate, angulararia, angularmessages, angularmd])
```
Comment (gbtec-ext-jorgevillar) [inline src/app/app.module.js]: Use camel case for `angularanimate, angulararia, angularmessages, angularmd`

File: src/app/shared/services/latestSearchesUtil.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        this.$window = $window;
+        this.latestSearches = latestSearches;
+    }
+
+    getLatestSearches() {
+        return this.latestSearches = JSON.parse(this.$window.localStorage.getItem('latestSearches')) || [];
+    };
+
+    addCity(city) {
+        if (city == undefined || city == null || city.trim() === '' || this.latestSearches.includes(city)) {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/shared/services/latestSearchesUtil.js]: You can do this check `city == undefined || city == null || city.trim() === ''` with this `!city  || city.trim() === ''`

And be careful with this solution, Jennifer Null and her family doesn't like it. 
https://gizmodo.com/it-really-sucks-to-be-named-jennifer-null-1767159091

File: src/app/shared/services/latestSearchesUtil.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            return;
+        }
+
+        if (city === this.latestSearches[0]) {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/shared/services/latestSearchesUtil.js]: isn't this check redundant with the `this.latestSearches.includes(city)` of the previous 'if'?

File: src/app/shared/services/latestSearchesUtil.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+export default class LatestSearchesUtil {
+    constructor($window, latestSearches) {
+        this.$window = $window;
```
Comment (eloydotco) [inline src/app/shared/services/latestSearchesUtil.js]: `window` object is already available, it's not needed to inject it

File: src/app/shared/services/latestSearchesUtil.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        this.latestSearches = latestSearches;
+    }
+
+    getLatestSearches() {
+        return this.latestSearches = JSON.parse(this.$window.localStorage.getItem('latestSearches')) || [];
```
Comment (eloydotco) [inline src/app/shared/services/latestSearchesUtil.js]: `localStorage` is also available as "global object"

File: src/app/shared/services/openWeatherMaps.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+            url: URL,
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        return this.fiveDayForecast = this.$http(request);
```
Comment (gbtec-ext-jorgevillar) [inline src/app/shared/services/openWeatherMaps.js]: You can use the shortcut version:
https://code.angularjs.org/1.5.11/docs/api/ng/service/$http#get

File: src/app/shared/services/openWeatherMaps.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
```
Comment (gbtec-ext-jorgevillar) [inline src/app/shared/services/openWeatherMaps.js]: Same as above

**Review by eloydotco — CHANGES_REQUESTED** [review]

File: src/app/shared/services/openWeatherMaps.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    }
```
Comment (gbtec-ext-jorgevillar) [inline src/app/shared/services/openWeatherMaps.js:43]: In this functions, do you want to return the response of the request? If that, you are failing :)

**Review by gbtec-ext-jorgevillar — CHANGES_REQUESTED** [review]

File: src/app/shared/services/openWeatherMaps.js (hunk 8 above)
Comment (jLagosCarrera) [reply src/app/shared/services/openWeatherMaps.js:43]: No, Im returning the promise

**Comment by jLagosCarrera** [comment]
Requested changes were made

File: src/app/app.module.js [hunk 9]
```diff
@@ -1,17 +1,27 @@
...
 import angular from 'angular';
-import angularmd from 'angular-material';
-import angularanimate from 'angular-animate';
-import angulararia from 'angular-aria';
-import angularmessages from 'angular-messages';
+import angularMd from 'angular-material';
+import angularAnimate from 'angular-animate';
+import angularAria from 'angular-aria';
+import angularMessages from 'angular-messages';
+import 'angular-material/angular-material.scss';
```
Comment (eloydotco) [inline src/app/app.module.js:7]: it's a good practice add and empty line between different imports blocks (I mean , by type, libraries, styles, services, etc), just for readability

File: src/app/shared/services/openWeatherMaps.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        const params = {
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        //return daily forecast promise
```
Comment (eloydotco) [inline src/app/shared/services/openWeatherMaps.js]: I think here there is something missing...

File: src/app/shared/services/openWeatherMaps.js (hunk 8 above)
Comment (eloydotco) [reply src/app/shared/services/openWeatherMaps.js:43]: I am with Jorge, you can return `this.$http.get(URL, params);` or first assign the value to `this.fiveDayForecast` and then return it, but not both.
Additionally, on the second option, `this.fiveDayForecast` will contain the promise, not the value, a better approach can be
````
return this.$http.get(URL, params).then(data => {
    this.fiveDayForecast = data; // maybe data need some transformation before assign it...
});

**Review by eloydotco — CHANGES_REQUESTED** [review]

**Comment by jLagosCarrera** [comment]
Requested changes were made

**Review by eloydotco — APPROVED** [review]

**Review by gbtec-ext-jorgevillar — APPROVED** [review]

File: src/app/shared/services/openWeatherMaps.js (hunk 8 above)
Comment (diegopf) [reply src/app/shared/services/openWeatherMaps.js:43]: what about async / await? :)

File: src/app/shared/services/openWeatherMaps.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+            .then((response) => {
+                this.currentForecast = response;
+            })
+            .catch((response) => {
+                this.latestError = response;
+            });
+    };
+
+    getFiveDayForecastCity(city) {
+        const URL = 'https://api.openweathermap.org/data/2.5/forecast';
```
Comment (diegopf) [inline src/app/shared/services/openWeatherMaps.js]: You should create a baseApi with 'https://api.openweathermap.org/data/2.5/' and build each endpoint url by concatenating the baseApi with the specific part

File: src/app/shared/services/openWeatherMaps.js [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            params: {
+                q: city,
+                units: 'metric',
+                lang: 'en',
+                appid: process.env.OPENWEATHERMAP_API_KEY
+            }
+        };
+
+        try {
+            return await this.$http.get(`${this.baseURL}/forecast`, params).then((response) => {
```
Comment (diegopf) [inline src/app/shared/services/openWeatherMaps.js]: what? Don't mix await with resolving the promise via then.

**Review by diegopf — CHANGES_REQUESTED** [review]

File: src/app/shared/services/openWeatherMaps.js (hunk 12 above)
Comment (jLagosCarrera) [reply src/app/shared/services/openWeatherMaps.js]: Havent finished yet, sorry for commiting!

**Comment by jLagosCarrera** [comment]
All requested changes are done

File: src/app/home-page/img/background.svg [hunk 13]
```diff
@@ -0,0 +1 @@
...1.51"/><path d="M387,843L538,687L374,694Z" fill="#369fc4" stroke="#369fc4" stroke-width="1.51"/><path d="M1003,2245L1158,2384L1137,2221Z" fill="#328daf" stroke="#328daf" stroke-width="1.51"/><path d="M1320,2386L1002,2394L1608,2398Z" fill="#3688af" stroke="#3688af" stroke-width="1.51"/><path d="M3289,2377L3299,2245L3154,2245Z" fill="#307c9f" stroke="#307c9f" stroke-width="1.51"/><path d="M3595,-85L3449,-219L3444,-72Z" fill="#2f6f93" stroke="#2f6f93" stroke-width="1.51"/><path d="M3306,-218L3604,-231L3136,-224Z" fill="#2f779a" stroke="#2f779a" stroke-width="1.51"/><path d="M3136,-224L3604,-231L2699,-237Z" fill="#3081a3" stroke="#3081a3" stroke-width="1.51"/><path d="M374,694L538,687L530,553Z" fill="#31a6c7" stroke="#31a6c7" stroke-width="1.51"/><path d="M381,537L374,694L530,553Z" fill="#30a5c4" stroke="#30a5c4" stroke-width="1.51"/><path d="M528,79L524,244L705,219Z" fill="#308cad" stroke="#308cad" stroke-width="1.51"/><path d="M528,79L705,219L681,77Z" fill="#308aab" stroke="#308aab" stroke-width="1.51"/><path d="M398,1006L386,1143L523,1157Z" fill="#4094c2" stroke="#4094c2" stroke-width="1.51"/><path d="M241,1009L386,1143L398,1006Z" fill="#3f8fbe" stroke="#3f8fbe" stroke-width="1.51"/
```
Comment (eloydotco) [inline src/app/home-page/img/background.svg]: All assets/resources like images should be on a folder on the project root folder

**Review by eloydotco — APPROVED** [review]

**Review by diegopf — APPROVED** [review]

**Pull request merged**